### ISSUE TITLE
Bug: cross_entropy_high_precision uses float32 instead of float64

### ISSUE THREAD

**Issue by AlexShaooo** [mechanistic-interpretability-grokking/progress-measures-paper#2]
## Description
The cross_entropy_high_precision function casts logits to torch.float32 despite the comment explicitly stating it should use float64 to avoid underflow issues.

## Location
- progress-measures-paper/helpers.py
```python
logprobs = F.log_softmax(logits.to(torch.float32), dim=-1)  # ← should be float64
``` 

## Expected Behavior
Based on the comment directly above this line:

Cast logits to float64 because log_softmax has a float32 underflow on overly confident data and can only return multiples of 1.2e-7

The cast should be to torch.float64, not torch.float32.

## Suggested Fix
```diff
- logprobs = F.log_softmax(logits.to(torch.float32), dim=-1)
+ logprobs = F.log_softmax(logits.to(torch.float64), dim=-1)
```

## Impact
This bug undermines the entire purpose of the high-precision function. As noted in the paper, reduced precision in loss calculations causes gradients to round to zero, leading to loss spikes and slingshot events:

<img width="1554" height="1134" alt="Image" src="https://github.com/user-attachments/assets/a3a35f60-c8f0-4af6-8b41-52659e2532d5" />
